numpy/core/numeric.py

```diff
@@ -1544,17 +1542,59 @@ def rollaxis(a, axis, start=0):
     return a.transpose(axes)
 
 
-def _validate_axis(axis, ndim, argname):
+def normalize_axis_tuple(axis, ndim, argname=None, allow_duplicate=False):
+    """
+    Normalizes a tuple of axes, such that they are  valid positive indices
+    into the shape of array with `ndim` dimensions. Raises an AxisError with an
+    appropriate message if this is not possible.
+
+    Handles shorthands such as ``1`` and converts it to ``(1,)``.
+
+    By default, disallows duplicates
+
+    Used internally by multi-axis-checking logic.
+
+    .. versionadded:: 1.13.0
+
+    Parameters
+    ----------
+    axis : int, tuple of int, or list of int
+        The un-normalized index or indices of the axis.
```

Comment:
The explanation should have a full stop at the end (here and below); I think this actually matters for sphinx rendering...